File: Counting_Sort/Counting_Sort.php
<?php
$list = array(3, 4, 5, 4, 7);
//Counting sort Function 
function countingsort($list , $largest,  $arrsize )
{
            $size = $largest + 1;

            $b= new SplFixedArray(10000);
            for($i=0 ; $i<$arrsize;$i++)
            {
                 $val = $list[$i];
                 $b[$val] =   $b[$val]+1;
            }
           $l = 0; 
           $r=0;
             
           $c= array(0,0,0,0,0);
            for ($j = 0; $l < $size; $j++)
            {


                while ($b[$l] != 0)
                {
                    $c[$r] = $l;
                    $r++;
                    $b[$l]=$b[$l]-1;

                }
                $l++;
            }
         
         for ( $k=0 ; $k<$arrsize;$k++)   
         echo $c[$k];

}
//Calling Countingsort function 
countingsort($list , 7 , 5);
// output 34457
?>

File: Counting_Sort/Counting_Sort.go
/*
                    Counting Sort
           ------------------------------------		
    Counting sort is a sorting technique based on keys between a specific range.
    It works by counting the number of objects having distinct key values (kind of hashing).
    Then doing some arithmetic to calculate the position of each object in the output sequence.
*/		   

package main
import "fmt"

const MaxUint = ^uint(0)
const MaxInt = int(MaxUint >> 1)
const MinInt = - MaxInt - 1

func countingSort(list []int) {
    maxNumber := MinInt
    minNumber := MaxInt
    for _, v := range list {
        if v > maxNumber {
            maxNumber = v
        }
        if v < minNumber {
            minNumber = v
        }
    }

    count := make([]int, maxNumber - minNumber + 1)

    for _, x := range list {
        count[x - minNumber]++
    }
    index := 0
    for i, c := range count {
        for ; c > 0; c-- {
            list[index] = i + minNumber
            index++
        }
    }
}

func main() {
    var x int
    fmt.Printf("Enter the size of the array : ")
    fmt.Scan(&x)
    fmt.Printf("Enter the elements of the array : ")
    a := make([]int, x)
    for i := 0; i < x; i++ {
        fmt.Scan(&a[i])
    }
    list := a    // A copy of the array 'a' is assigned to 'list' 
    fmt.Println("before sorting", list)
    countingSort(list)
    fmt.Println("after sorting", list)
}

/*
Sample Input :
    Enter the size of the array : 4                                                                                              
    Enter the elements of the array : 12 29 10 78

Sample Output :
    After sorting [10 12 29 78]	
*/

